### PR TITLE
Add support for uploading/downloading to/from subdirectories

### PR DESCRIPTION
**Pull request by moparisthebest**
The original code only supported /, this adds support for subdirectories like /up/ or whatever, with separate config for put/get urls.  It will work with old configurations without problems and as before.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -183,7 +184,7 @@ def do_PUT(self):
         global files
         global files_lock
         global config
-        path = normalize_path(self.path[1:])
+        path = normalize_path(self.path, config['put_sub_url_int'])
```
Comment (SamWhited) [inline server.py]: I don't see this in the config example, or any documentation for what it does?

Comment (SamWhited) [inline server.py]: Oops, nevermind, I misunderstood.

File: server.py [hunk 2]
```diff
@@ -268,6 +269,18 @@ class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
                             format='%(asctime)-24s %(levelname)-8s %(message)s',
                             filename=args.logfile)
 
+    try:
+        config['get_sub_url'] = config.get('get_sub_url', '/')
+        config['put_sub_url'] = config.get('put_sub_url', '/')
+        config['get_sub_url_int'] = int(len(config['get_sub_url']))
```
Comment (SamWhited) [inline server.py]: `get_sub_url_len` or something might be better? the datatype (int) doesn't really tell me what this is suposed to be. Also, no need to wrap in `int()`, just `len(config[...` will do.

Comment (andersruneson) [inline server.py]: I'm fairly sure that the current code supports subdirs in the get-url??

**Pull request merged**